client/ayon_deadline/lib.py

```diff
@@ -43,3 +48,368 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
+    OverrideJobCleanupType: Optional[str] = field(default=None)
+
+    # Scheduling
+    ScheduledType: Optional[str] = field(
+        default=None)  # Default: None (<None/Once/Daily/Custom>)
+    ScheduledStartDateTime: Optional[str] = field(
+        default=None)  # <dd/MM/yyyy HH:mm>
+    ScheduledDays: Optional[int] = field(default=1)  # Default: 1
+    JobDelay: Optional[str] = field(default=None)  # <dd:hh:mm:ss>
+    Scheduled: Optional[str] = field(
+        default=None)  # <Day of the Week><Start/Stop>Time=<HH:mm:ss>
+
+    # Scripts
+    PreJobScript: Optional[str] = field(default=None)  # Default: blank
+    PostJobScript: Optional[str] = field(default=None)  # Default: blank
+    PreTaskScript: Optional[str] = field(default=None)  # Default: blank
+    PostTaskScript: Optional[str] = field(default=None)  # Default: blank
+
+    # Event Opt-Ins
+    EventOptIns: Optional[str] = field(
+        default=None)  # Default blank (comma-separated list)
+
+    # Environment
+    EnvironmentKeyValue: Any = field(
+        default_factory=partial(DeadlineKeyValueVar, "EnvironmentKeyValue"))
+    IncludeEnvironment: Optional[bool] = field(default=False)  # Default: false
+    UseJobEnvironmentOnly: Optional[bool] = field(default=False)  # Default: false
+    CustomPluginDirectory: Optional[str] = field(default=None)  # Default blank
+
+    # Job Extra Info
+    ExtraInfo: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "ExtraInfo"))
+    ExtraInfoKeyValue: Any = field(
+        default_factory=partial(DeadlineKeyValueVar, "ExtraInfoKeyValue"))
+
+    OverrideTaskExtraInfoNames: Optional[bool] = field(
+        default=False)  # Default false
+
+    TaskExtraInfoName: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "TaskExtraInfoName"))
+
+    OutputFilename: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename"))
+    OutputFilenameTile: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename{}Tile"))
+    OutputDirectory: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputDirectory"))
+
+    AssetDependency: str = field(
+        default_factory=partial(DeadlineIndexedVar, "AssetDependency"))
+
+    TileJob: bool = field(default=False)
+    TileJobFrame: int = field(default=0)
+    TileJobTilesInX: int = field(default=0)
+    TileJobTilesInY: int = field(default=0)
+    TileJobTileCount: int = field(default=0)
+
+    MaintenanceJob: bool = field(default=False)
+    MaintenanceJobStartFrame: int = field(default=0)
+    MaintenanceJobEndFrame: int = field(default=0)
+
+
+@dataclass
+class AYONDeadlineJobInfo(DeadlineJobInfo):
+    """Contains additional AYON variables from Settings for internal logic."""
+
+    # AYON custom fields used for Settings
+    UsePublished: Optional[bool] = field(default=None)
+    UseAssetDependencies: Optional[bool] = field(default=None)
+    UseWorkfileDependency: Optional[bool] = field(default=None)
```

Comment:
I'm a bit confused by these. Why do we need to store these on a "Job Info" object?

They are either "Job Info" and hence deadline related.
Or they are AYON info... but not Deadline Job Info related because deadline would not support it, right?

Or how is this intended?

Reply:
The idea is to clearly separate internal fields which serve some additional logic. 
We have for example `use_asset_dependencies` as checkbox, if one should/shouldn't use them. But DLJI has only `AssetDependency` field which should (probably) contain list of assets on which the job is dependent (which is probably broken, by the way, as there is simple concatenation there, I would expect usage of some delimiter to concatenate string values).
Dtto with `machine_list`.

Reply:
Hmm - I see. I think we should fix those values BEFORE passing it to the info objects actually - so that as we access them as "job info" they are the types we expect it to be for Deadline. Much less hassle.

Regarding the AYON-specific keys my point was that I didn't expect them to be on "Deadline Job Info" at all - but somewhere separate from it.

Reply:
I dont have list of assets to be dependent in collector phase. 
I think I could clean it up a bit, but I feel there will be some additional fields that would need to be passed through to the submission phase either way.